awswrangler/s3/_write_dataset.py

```diff
@@ -190,4 +258,27 @@ def _to_dataset(
         )
     _logger.debug("paths: %s", paths)
     _logger.debug("partitions_values: %s", partitions_values)
+    if (table_type == "GOVERNED") and (table is not None) and (database is not None):
+        add_objects: List[Dict[str, Any]] = _build_table_objects(
+            paths, partitions_values, use_threads=use_threads, boto3_session=boto3_session
+        )
+        try:
+            if add_objects:
+                _update_table_objects(
+                    catalog_id=catalog_id,
+                    database=database,
+                    table=table,
+                    transaction_id=transaction_id,  # type: ignore
+                    add_objects=add_objects,
+                    boto3_session=boto3_session,
+                )
+                if commit_trans:
+                    commit_transaction(transaction_id=transaction_id, boto3_session=boto3_session)  # type: ignore
+        except Exception as ex:
+            _logger.debug("Aborting transaction with ID: %s.", transaction_id)
+            if transaction_id:
+                abort_transaction(transaction_id=transaction_id, boto3_session=boto3_session)
+            _logger.error(ex)
+            raise
+
```

Comment:
If table_type is "GOVERNED", add S3 objects to the Lake Formation engine for the specific table. Abort the transaction if error